operators/pkg/controller/elasticsearch/certificates/transport/pod_secret_test.go

```diff
@@ -55,21 +70,38 @@ func TestEnsureTransportCertificateSecretExists(t *testing.T) {
 				pod:   pod,
 			},
 			want: func(t *testing.T, secret *corev1.Secret) {
-				if assert.Contains(t, secret.Labels, LabelCertificateType) {
-					assert.Equal(t, secret.Labels[LabelCertificateType], LabelCertificateTypeTransport)
-				}
-				assert.Equal(t, "pod-certs", secret.Name)
+				// owner references are set upon creation, so ignore for comparison
+				expected := defaultPodSecretWith(func(s *corev1.Secret) {
+					s.OwnerReferences = secret.OwnerReferences
+				})
+				assert.Equal(t, expected, secret)
+			},
+		},
+		{
+			name: "should update an existing secret",
+			args: args{
+				c: k8s.WrapClient(fake.NewFakeClient(defaultPodSecretWith(func(secret *corev1.Secret) {
+					secret.ObjectMeta.UID = types.UID("42")
+				}))),
+				owner: es,
+				pod:   pod,
+			},
+			want: func(t *testing.T, secret *corev1.Secret) {
+				// UID should be kept the same
+				assert.Equal(t, defaultPodSecretWith(func(secret *corev1.Secret) {
+					secret.ObjectMeta.UID = types.UID("42")
+				}), secret)
 			},
 		},
 		{
-			name: "should not create a new secret if it already exists",
+			name: "should allow additional labels in the secret",
 			args: args{
-				c:     k8s.WrapClient(fake.NewFakeClient(preExistingSecret)),
+				c:     k8s.WrapClient(fake.NewFakeClient(defaultPodSecret)),
```

Comment:
not sure I understand how this is adding additional labels?

Reply:
Sorry, that got lost in some refactoring, added it back in :)